src/node/ledger_secrets.h

```diff
@@ -54,28 +54,38 @@ namespace ccf
 
     SpinLock lock;
     LedgerSecretsMap ledger_secrets;
-    std::optional<LedgerSecretsMap::iterator> commit_key_it = std::nullopt;
 
-    LedgerSecretsMap::iterator get_encryption_key_it(kv::Version version)
+    LedgerSecret get_secret_for_version(kv::Version version)
```

Comment:
Can we have some unit tests for this please?

Reply:
This is already tested via the encryptor's unit test which makes use of this class (see `encryptor_test.cpp`).